docs/basic-component.md

```diff
@@ -0,0 +1,266 @@
+# Basic component recipe
+
+Recipe for creating a basic component.
+
+- [Component structure](#component-structure)
+  - [Subcomponents](#subcomponents)
+  - [Hooks and Utils](#hooks-and-utils)
+  - [index.ts](#indexts-1)
+  - [Component.tsx](#componenttsx)
+  - [Styles](#styles)
+  - [Tests](#tests)
+- [Component.tsx template](#componenttsx-template)
+  - [Props](#props)
+  - [The component](#the-component)
+  - [Display name](#display-name)
+
+## Component structure
+
+Given the component `PancakeStack`, the component structure will follow this:
+
+```
+/* Component */
+src/
+  PancakeStack/
+    components/
+      - Pancake/
+        - index.ts
+        - Pancake.tsx
+        - Pancake.scss
+        - Pancake.spec.tsx
+      - Topping/
+        - index.ts
+        - Topping.tsx
+        - Topping.scss
+        - Topping.spec.tsx
+    hooks/
+      - index.ts
+      - useHook.ts
+      - useHook.spec.ts
+    utils/
+      - index.ts
+      - groupedFunctions.ts (eg. getters)
+      - groupedFunctions.spec.ts
+      - functionName.ts
+      - functionName.spec.ts
+    - _variables.scss
+    - index.ts
+    - PancakeStack.tsx
+    - PancakeStack.scss
+    - PancakeStack.spec.tsx
+```
+
```

Comment:
Hmm... I would actually think following on our patterns, it would be something like this:

```
typography/
    docs/
		- DataHeader.stories.tsx
		- DataHeader.stories.scss
		- Heading.stories.tsx
		- Paragraph.stories.tsx
		- SomeStoryHelperComponent.tsx
		- SomeStoryHelperComponent.scss
    src/
		DataHeader/
	        components/
	        hooks/
	        utils/
	        - _variables.scss
	        - index.ts
	        - DataHeader.tsx
	        - DataHeader.scss
	        - DataHeader.spec.tsx
        Heading/
			...
		Paragraph/
			...
    - index.ts
    - package.json
    ... other stuff
```

I think it will look less strange once the stories can actually live with their components, but in the meantime it's quite similar to the existing structure where the docs are detached.
 
I believe the `package.json` config I copied from the old template relies on `docs` being on the top level to be excluded.
But if it isn't tied into the top level, since we'd still need it in a `docs` directory in the meantime, it would look like this:
```
typography/
    src/
		DataHeader/
			docs/
				- DataHeader.stories.tsx
				- DataHeader.stories.scss
				- SomeStoryHelperComponent.tsx
				- SomeStoryHelperComponent.scss
	        components/
	...etc
```

Normally the top level `docs/` would be for documentation about the repo as opposed to component docs (which would live within the component directories), which is why the components should always be within the top level `src/` and not switched.
Might be easier to remember this: there is one `package` and multiple `components`, and `src/` belongs to the `package` :)

Reply:
Gotcha, so single src, top level docs for actual documentation not stories would produce something like this:
```
typography/
  docs/
    // markdown if relevant but not stories
  src/
    DataHeader/
      docs/
        - DataHeader.stories.tsx
      components/
      hooks/
      utils/
      - _variables.scss
      - index.ts
      - DataHeader.tsx
      - DataHeader.scss
      - DataHeader.spec.tsx
    otherComponent/
  - index.ts
  ...etc
```

Reply:
Ideally I think it would be good for it to look like this (without the need for a `docs/` in the component), but it needs thought on how to deal with those helper things.

```
typography/
  docs/
    // markdown if relevant but not stories
  src/
    DataHeader/
      components/
      hooks/
      utils/
      - _variables.scss
      - index.ts
      - DataHeader.tsx
      - DataHeader.scss
      - DataHeader.spec.tsx
      - DataHeader.stories.tsx
    otherComponent/
  - index.ts
  ...etc
```

But maybe the in house `docs/` is a good middle ground 🤔 

Or perhaps we have a `__docs/` or `__stories/` (`__something` is a common pattern I've to signify internal things) which you only put in when you need components beyond the `*.stories.*` files.

```
typography/
  docs/
    // markdown if relevant but not stories
  src/
    DataHeader/
      __docs/
        - SomeStoryHelperComponent.tsx
        - SomeStoryHelperComponent.scss
      components/
      hooks/
      utils/
      - _variables.scss
      - index.ts
      - DataHeader.tsx
      - DataHeader.scss
      - DataHeader.spec.tsx
      - DataHeader.stories.tsx
      - DataHeader.stories.scss
    otherComponent/
  - index.ts
  ...etc
```

Reply:
Sounds good. There's probably a bigger conversation on how we want to handle dev docs outside of storybook, which would tie nicely into retooling how we split out our stories, but for the moment I think the original proposed pattern works 👍